src/manifold/src/constructors.cpp

```diff
@@ -231,13 +261,38 @@ Manifold Manifold::Sphere(float radius, int circularSegments) {
  * @param twistDegrees Amount to twist the top crossSection relative to the
  * bottom, interpolated linearly for the divisions in between.
  * @param scaleTop Amount to scale the top (independently in X and Y). If the
- * scale is {0, 0}, a pure cone is formed with only a single vertex at the top.
- * Note that scale is applied after twist.
- * Default {1, 1}.
+ * scale is {0, 0}, a pure cone is formed with only a single vertex at the
+ * top. Note that scale is applied after twist. Default {1, 1}.
  */
 Manifold Manifold::Extrude(const CrossSection& crossSection, float height,
                            int nDivisions, float twistDegrees,
                            glm::vec2 scaleTop) {
+  return ExtrudeTransforming(crossSection, height, nDivisions, twistDegrees,
+                             scaleTop);
+}
+
+/**
+ * Constructs a manifold from a set of polygons by extruding them along the
+ * Z-axis.
+ * Note that high twistDegrees with small nDivisions may cause
+ * self-intersection. This is not checked here and it is up to the user to
+ * choose the correct parameters.
+ *
+ * @param crossSection A set of non-overlapping polygons to extrude.
+ * @param height Z-extent of extrusion.
+ * @param nDivisions Number of extra copies of the crossSection to insert into
+ * the shape vertically; especially useful in combination with twistDegrees to
+ * avoid interpolation artifacts. Default is none.
+ * @param twistDegrees Amount to twist the top crossSection relative to the
+ * bottom, interpolated linearly for the divisions in between.
+ * @param scaleTop Amount to scale the top (independently in X and Y). If the
+ * scale is {0, 0}, a pure cone is formed with only a single vertex at the
+ * top. Note that scale is applied after twist. Default {1, 1}.
+ */
+Manifold Manifold::ExtrudeTransforming(const CrossSection& crossSection,
```

Comment:
Same problem - `isConvex` is not as trivial to test as you might think. If you're certain of your answer, you probably aren't doing it right.

Reply:
@elalish Possible mis-understanding here.  The isConvex flag is set by the caller, THE CALLER determines whether or not the input is convex.  For example, Cylinder, the circles involved OBVIOUSLY are convex, and thus the version of Cylinder I made sets that flag. The default is False, and will work just fine, but slower.

Reply:
@briansturgill The issue here is that very often users don't (exactly) know what they are doing, they often just copy some example code or read some comments about *HOW TO MAKE YOUR CODE 3 TIMES FASTER* and just set `isConvex = true`. In those cases our internal data structure will silently corrupt, then users will come and ask us why our library is outputting problematic meshes. This is why we are hesitant in adding this kind of `unchecked` APIs.

E.g. `convexity` parameter in openscad. A lot more complicated than just `isConvex`, but the idea is the same...